frontend/pages/hosts/ManageHostsPage/components/LabelFilterSelect/LabelFilterSelect.tsx

```diff
@@ -0,0 +1,175 @@
+  }
+
+  return (
+    <div className={"option-label"}>
+      {isPlatform && (
+        <img src={PLATFORM_TYPE_ICONS[data.display_text]} alt="" />
+      )}
+      <span>{labelText}</span>
+    </div>
+  );
+};
+
+const baseClass = "label-filter-select";
+
+interface ILabelFilterSelectProps {
+  labels: ILabel[];
+  selectedLabel: ILabel | null;
+  canAddNewLabels: boolean;
+  className?: string;
+  onChange: (labelId: ILabel) => void;
+  onAddLabel: () => void;
+}
+
+const LabelFilterSelect = ({
+  labels,
+  selectedLabel,
+  canAddNewLabels,
+  className,
+  onChange,
+  onAddLabel,
+}: ILabelFilterSelectProps) => {
+  const [labelQuery, setLabelQuery] = useState("");
+
+  // we need the Select to be a controlled component to enable our label input
+  // to work correctly. shouldOpenMenu now becomes our single source of truth if
+  // we want the menu to render open or closed.
+  const [shouldOpenMenu, setShouldOpenMenu] = useState(false);
+  const isLabelSearchInputFocusedRef = useRef(false);
+  const selectRef = useRef<
+    SelectInstance<ILabel | IEmptyOption, false, IGroupOption>
+  >(null);
+
+  const options = useMemo(() => createDropdownOptions(labels, labelQuery), [
+    labels,
+    labelQuery,
+  ]);
+
+  const handleChange = (option: ILabel | IEmptyOption | null) => {
+    if (option === null) return;
+    if ("type" in option) {
+      setShouldOpenMenu(false);
+      setLabelQuery("");
+      selectRef.current?.blur();
+      onChange(option);
+    }
+  };
+
+  const handleLabelQueryChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    // We need to stop the key presses propagation to prevent the dropdown from
+    // picking up keypresses.
+    event.stopPropagation();
+    setLabelQuery(event.target.value);
+  };
+
+  const handleBlurSelect = () => {
+    if (!isLabelSearchInputFocusedRef.current) {
+      isLabelSearchInputFocusedRef.current = false;
+      setShouldOpenMenu(false);
+    }
+  };
+
+  const handleFocusSelect = () => {
+    setShouldOpenMenu(true);
+  };
+
+  const handleClickLabelSearchInput = () => {
+    isLabelSearchInputFocusedRef.current = true;
+  };
+
+  const handleBlurLabelSearchInput = () => {
+    isLabelSearchInputFocusedRef.current = false;
+    setShouldOpenMenu(false);
+  };
+
+  const getOptionLabel = (option: ILabel | IEmptyOption) => {
+    if ("display_text" in option) {
+      return option.display_text;
+    }
+    return option.label;
+  };
+
+  const getOptionValue = (option: ILabel | IEmptyOption) => {
+    if ("id" in option) {
+      return option.id.toString();
+    }
+    return option.label;
+  };
+
+  const classes = classnames(baseClass, className);
+
+  return (
+    <Select<ILabel | IEmptyOption, false, IGroupOption>
+      ref={selectRef}
+      name="input-filter-select"
+      options={options}
+      className={classes}
+      classNamePrefix={baseClass}
+      defaultMenuIsOpen={false}
+      placeholder={"Filter by operating system or label"}
```

Comment:
@ghernandez345 heads up, it looks like I forgot to update Figma. 

We'd like the placeholder to read "Filter by platform or label" instead of "Filter by operating system or label."

Reply:
@noahtalerman sounds good. I am going to merge this current PR  and include this change in another PR im about to do to remove the code no longer used for the host sidebar